src/Microsoft.Framework.Runtime/ApplicationHostContext.cs

```diff
@@ -44,6 +47,14 @@ public class ApplicationHostContext
             ProjectDepencyProvider = new ProjectReferenceDependencyProvider(ProjectResolver);
             var unresolvedDependencyProvider = new UnresolvedDependencyProvider();
 
+            IEnumerable<IDependencyProvider> dependencyProviders = new IDependencyProvider[] {
```

Comment:
If you want to avoid copying this several times move it to a separate function. This way you will avoid this kind of weird pattern where you override the source for iterator with a computation.